AISKU/Tests/perfTests.ts

```diff
@@ -0,0 +1,363 @@
+/// <reference path='./TestFramework/Common.ts' />
+import {
+  ApplicationInsights,
+  IApplicationInsights,
+} from "../src/applicationinsights-web";
+import { Sender } from "@microsoft/applicationinsights-channel-js";
+import {
+  NotificationManager,
+  IPerfEvent,
+  PerfManager,
+} from "@microsoft/applicationinsights-core-js";
+import {
+  Event,
+  Trace,
+  Exception,
+  Metric,
+  PageView,
+  PageViewPerformance,
+  RemoteDependencyData,
+} from "@microsoft/applicationinsights-common";
+import { EventValidator } from "./TelemetryValidation/EventValidator";
+import { TraceValidator } from "./TelemetryValidation/TraceValidator";
+import { ExceptionValidator } from "./TelemetryValidation/ExceptionValidator";
+import { MetricValidator } from "./TelemetryValidation/MetricValidator";
+import { PageViewPerformanceValidator } from "./TelemetryValidation/PageViewPerformanceValidator";
+import { PageViewValidator } from "./TelemetryValidation/PageViewValidator";
+import { RemoteDepdencyValidator } from "./TelemetryValidation/RemoteDepdencyValidator";
+
+export class PerfTests extends TestClass {
+  private readonly _instrumentationKey = "b7170927-2d1c-44f1-acec-59f4e1751c11";
+
+  private _ai: IApplicationInsights;
+
+  // Sinon
+  private errorSpy: SinonSpy;
+  private successSpy: SinonSpy;
+  private loggingSpy: SinonSpy;
+
+  private delay = 100;
+  private _notificationManager;
+  private _sendNotifications;
+
+  public testInitialize() {
+    try {
+      this.useFakeServer = false;
+      (sinon.fakeServer as any).restore();
+      this.useFakeTimers = false;
+      this.clock.restore();
+
+      // Setup Notification Listener
+      this._sendNotifications = [];
+      this._notificationManager = new NotificationManager();
+      this._notificationManager.addNotificationListener({
+        perfEvent: (perfEvent: IPerfEvent): void => {
+          this._sendNotifications.push(perfEvent);
+        },
+      });
+
+      const init = new ApplicationInsights({
```

Comment:
As we are creating specific tests for initialization, we should move this into the specific tests, to allow for updating of the configurations